### PR TITLE
Move bookmarks from EmailUser to UserProfile

### PR DESCRIPTION
**Pull request by Pomax**
Addresses https://github.com/mozilla/network-pulse-api/issues/133

This PR sets up a UserProfile model, and moves user `bookmarks` over from the EmailUser model to this UserProfile model instead. This will let us both start padding out the UserProfile, and clean up the EmailUser to the point where we can hopefully replace it with a Django-managed user instead.

How to test this PR:

1. follow the setup instructions in the README for making sure you can use/log in
2. grab this database file: [db.tar.gz](https://github.com/mozilla/network-pulse-api/files/1201641/db.tar.gz) and unpack it to the root. This is a database with migrations applied equal to the pre-this-PR codebase, with a bare minimum of faves stored
3. run `manage migrate`, then `manage test`. The migration will likely ask you to delete the stale database users_userbookmarks: that is fine. Running `manage test` should show 41 tests passing.
4. run `manage runserver` and hit up your http://test.example.com:8000/admin and have a look at the profile/bookmark data, as well as what's in the emailuser entries. It should all look kosher.

### CODE REVIEW

File: pulseapi/profiles/models.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    bookmarks = models.ManyToManyField(
+        'entries.Entry',
+        through='profiles.UserBookmarks'
+    )
+
+    def __str__(self):
+        return 'profile for {}'.format(self.user.email)
+
+
+class UserBookmarks(models.Model):
```
Comment (gideonthomas) [inline pulseapi/profiles/models.py]: nit: probably need a `Meta` class to specify the verbose name as `Bookmark` so it doesn't add the extra `s` - `Bookmarkss`

File: pulseapi/users/admin.py [hunk 2]
```diff
@@ -3,33 +3,46 @@
...
+            id=profile.id,
+        )
+
+        return format_html(html)
+
+    def bookmarks(self, instance):
+        """
+        Show all bookmarked entries as a string of titles. In the future we should make them links.
+        """
+        profile = UserProfile.objects.get(user=intance)
```
Comment (gideonthomas) [inline pulseapi/users/admin.py]: typo: `instance`

File: pulseapi/users/admin.py [hunk 3]
```diff
@@ -3,33 +3,46 @@
...
 class EmailUserAdmin(admin.ModelAdmin):
     """
     Show a list of entries a user has submitted in the EmailUser Admin app
     """
-    fields = ('password', 'last_login', 'email', 'name', 'entries','bookmarks', 'is_staff', 'is_superuser')
-    readonly_fields = ('entries','bookmarks')
-
-    # this allows us to create/edit/delete/etc bookmarks:
-    inlines = [ UserBookmarksInline ]
+    fields = ('password', 'last_login', 'email', 'name', 'is_staff', 'is_superuser', 'profile', 'entries','bookmarks', )
```
Comment (gideonthomas) [inline pulseapi/users/admin.py]: so we still show bookmarks in the EmailUser, not in the UserProfile?

File: pulseapi/profiles/admin.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from django.contrib import admin
+from .models import UserProfile, UserBookmarks
+
+class UserProfileAdmin(admin.ModelAdmin):
+    """
+    Show a list of entries a user has submitted in the EmailUser Admin app
```
Comment (gideonthomas) [inline pulseapi/profiles/admin.py]: We'll change the comments in a future pr when we add the actual fields to a user profile?

File: pulseapi/users/admin.py (hunk 3 above)
Comment (Pomax) [inline pulseapi/users/admin.py]: Both. It's useful for admins to see the bookmark count in the user view (although I took out the inline, there is no reason to see each bookmark as editable entity...)

File: pulseapi/profiles/admin.py (hunk 4 above)
Comment (Pomax) [inline pulseapi/profiles/admin.py]: we will, but that doesn't make this the right comment - fixing to say it shows the profile-associated user, if there is one set.

**Pull request merged**